### ISSUE TITLE
[Bug] Migration gotrue

### ISSUE THREAD

**Issue by ducdao-cdh** [AppFlowy-IO/AppFlowy-Cloud#823]
**Describe the bug**
I am trying to initialize gotrue with docker. Here are the error details and the service I am using:

![image](https://github.com/user-attachments/assets/c7c4708b-0569-4aa6-8244-0a335814b3f2)
![image](https://github.com/user-attachments/assets/1d9feca6-98e7-4053-a34a-336a54c0dd7c)

![image](https://github.com/user-attachments/assets/6fcb15c4-2a50-44c8-9be8-a96182c988dd)
![image](https://github.com/user-attachments/assets/5043e2ba-aad4-4ce1-a378-b4dac513a286)

I am using super admin account for db, so I think this is not a permissions issue

**Comment by DenuxPlays** [comment]
Same error

**Comment by speed2exe** [comment]
@ducdao-cdh due to historic reason, you might need to run the sql commands in `migrations/before` before starting the gotrue service

**Comment by ducdao-cdh** [comment]
![image](https://github.com/user-attachments/assets/e6f79594-cf52-4d58-b667-dd4fa90a867b)

@speed2exe I still get this error after running ```migrations/before/20230312043000_supabase_auth.sql```

**Comment by ducdao-cdh** [comment]
![image](https://github.com/user-attachments/assets/d8d55bd7-6492-4c59-808a-560d0bb67427)

After creating user "postgres" and granting permissions I got this error

**Comment by speed2exe** [comment]
@ducdao-cdh Looking at the logs I couldn't pinpoint what went wrong. Perhaps you can try with the most basic configuration, make sure it works, then slowly introduce changes like database name.

**Comment by ducdao-cdh** [comment]
ok, let me try. Thanks u !

**Comment by almereyda** [comment]
Possible duplicate of #839 or vice versa.

Instead of adding said `postgres` user, we could also default to replacing it with `supabase_auth_admin` in the build process of the images.

**Comment by y2141-7991** [comment]
When i `cargo run` i got this error

```
postgres-1   | 2024-10-08 20:16:44.784 UTC [39] ERROR:  relation "identities" does not exist at character 210
postgres-1   | 2024-10-08 20:16:44.784 UTC [39] STATEMENT:  SELECT identities.created_at, identities.email, identities.id, identities.identity_data, identities.last_sign_in_at, identities.provider, identities.provider_id, identities.updated_at, identities.user_id FROM identities AS identities WHERE email = $1
gotrue-1     | {"component":"api","error":"unable to find identity by email for duplicates: unable to fetch records: ERROR: relation \"identities\" does not exist (SQLSTATE 42P01)","level":"error","method":"POST","msg":"500: Database error finding user","path":"/signup","referer":"appflowy-flutter://","remote_addr":"172.21.0.1","request_id":"bf29d213-7be5-48f3-b745-b5e27c5206cc","time":"2024-10-08T20:16:44Z"}
```
and error at this endpoint

![image](https://github.com/user-attachments/assets/6c690b19-950e-4d62-b891-ed05da85ddd1)

**Comment by y2141-7991** [comment]
When i `cargo run` i got this error

```
postgres-1   | 2024-10-08 20:16:44.784 UTC [39] ERROR:  relation "identities" does not exist at character 210
postgres-1   | 2024-10-08 20:16:44.784 UTC [39] STATEMENT:  SELECT identities.created_at, identities.email, identities.id, identities.identity_data, identities.last_sign_in_at, identities.provider, identities.provider_id, identities.updated_at, identities.user_id FROM identities AS identities WHERE email = $1
gotrue-1     | {"component":"api","error":"unable to find identity by email for duplicates: unable to fetch records: ERROR: relation \"identities\" does not exist (SQLSTATE 42P01)","level":"error","method":"POST","msg":"500: Database error finding user","path":"/signup","referer":"appflowy-flutter://","remote_addr":"172.21.0.1","request_id":"bf29d213-7be5-48f3-b745-b5e27c5206cc","time":"2024-10-08T20:16:44Z"}
```
and error at this endpoint



> @y2141-7991 seems like database configuration went wrong. may I ask what steps did you take to trigger this error?



> @y2141-7991 seems like database configuration went wrong. may I ask what steps did you take to trigger this error?

@speed2exe 
I added env to docker compose
`docker compose --file docker-compose-dev.yml up`
`sqlx database create`

I created schema auth manually
`
CREATE SCHEMA auth;
`
When i re-run docker i got this error 
```
{"args":[0.014014875],"component":"pop","level":"info","msg":"%.4f seconds","time":"2024-08-20T19:06:21Z"}
{"level":"fatal","msg":"running db migrations: error executing migrations/20240729123726_add_mfa_phone_config.up.sql, sql: do $$ begin\n    alter type auth.factor_type add value 'phone';\nexception\n    when duplicate_object then null;\nend $$;\n\n\nalter table auth.mfa_factors add column if not exists phone text unique default null;\nalter table auth.mfa_challenges add column if not exists otp_code text null;\n\n\ncreate unique index if not exists unique_verified_phone_factor on auth.mfa_factors (user_id, phone);\n: ERROR: type \"auth.factor_type\" does not exist (SQLSTATE 42704)","time":"2024-08-20T19:06:21Z"}
```

https://github.com/supabase/auth/issues/1729
and i followed this step in this issue
```
do $$ begin
    create type auth.factor_type as enum('totp', 'webauthn');
exception
    when duplicate_object then null;
end $$;
```

After this, i run command `cargo run` and I got the error as above

**Comment by speed2exe** [comment]
@y2141-7991 Are you trying to set up local development based on https://github.com/AppFlowy-IO/AppFlowy-Cloud/blob/main/README.md#-development?

**Comment by y2141-7991** [comment]
@speed2exe Yes, i am

**Comment by kevinleung56** [comment]
I've got the same issue

**Comment by chevdor** [comment]
Same issue, debugging the query, I see:
```
SELECT
	identities.created_at, identities.email, identities.id, identities.identity_data,
	identities.last_sign_in_at, identities.provider, identities.provider_id,
	identities.updated_at, identities.user_id
FROM identities AS identities <...>;
```

This query fails on my DB.

However the following works:
```
SELECT
	identities.created_at, identities.email, identities.id, identities.identity_data,
	identities.last_sign_in_at, identities.provider, identities.provider_id,
	identities.updated_at, identities.user_id
FROM auth.identities AS identities <...>;                                                   <<<< DIFF HERE
```

**Comment by speed2exe** [comment]
@almereyda For development wise, you shouldn't be creating the schema manually. assuming that you are following instructions, you should be doing the following instead:
```bash
sqlx database create
sqlx migrate run
```

**Comment by speed2exe** [comment]
@chevdor this seems looks like you did not specify the database name in the `DATABASE_URL`.

**Comment by chevdor** [comment]
@speed2exe I did, otherwise it would not connect at all.
I also initially used some `appflowy` DB and read about the issues related to the DB not being `postgres` so I reverted to that.

I do have the following defined:
- `DATABASE_URL`
- `APPFLOWY_DATABASE_URL`
- `APPFLOWY_AI_DATABASE_URL`
- `APPFLOWY_HISTORY_DATABASE_URL`
- `APPFLOWY_WORKER_DATABASE_URL`

The content is in the form:
```
postgres://postgres:somepass@svc-appflowy-postgresql:5432/postgres
```
`svc-appflowy-postgresql` being my K8S service for the db.
 
It is very redundant btw since they are all the same. Maybe those are not all required, not sure.

**Comment by speed2exe** [comment]
@chevdor This looks like a `gotrue` migration, you might need to replace the 'GOTRUE_DATABASE_URL' variable instead.

**Comment by chevdor** [comment]
I just tried and gotrue fails with `{"level":"fatal","msg":"Failed to load configuration: required key DATABASE_URL missing value","time":"2024-10-21T13:09:53Z"}`. So I think `DATABASE_URL` is correct.
Moreover, the gotrue migration actually seemed to run fine so it was definitely able to access the DB.

**Comment by speed2exe** [comment]
@chevdor In the docker-compose files, there's one line: `      - DATABASE_URL=${GOTRUE_DATABASE_URL}
`
`DATABASE_URL` is correct for gotrue, but you need to use the values in `GOTRUE_DATABASE_URL` in the `.env` files

**Comment by chevdor** [comment]
I am not using docker-compose, nor using the env. I am using K8S, and configmaps for env and secrets.

So my `GOTRUE_DATABASE_URL` is currently not set but that should not matter as I do set those I listed above.

In other words, I define `DATABASE_URL` directly as in an ENV as `DATABASE_URL=[postgres.....](postgres://postgres:somepass@svc-appflowy-postgresql:5432/postgres`

**Comment by chevdor** [comment]
Yes, and I do have it set. Beside the issue I mentioned 3d ago, all the other migrations did work fine, so the DB connection is fine.

**Comment by speed2exe** [comment]
@chevdor I see. It's probably something to do with default schema in postgresql not set to `auth`. You might be able to fix this by including `ALTER ROLE supabase_auth_admin SET search_path TO auth;` when you initialize postgres

**Comment by chevdor** [comment]
Yes, that sounds good. I will test that.

**Comment by chevdor** [comment]
Ok, I ran `ALTER ROLE postgres SET search_path TO auth;` and it seems to progress a little further.
All in all, gotrue is happy. AppFlowy now complains about migration 20230312043023 now failing:
```
Error: Failed to initialize application state: Failed to run migrations: while executing migration 20230312043023: error returned from database: policy "af_user_insert_policy" for table "af_user" already exists
```

I will leave it at that here since it does not look that my problem is anymore related to gotrue and I don't want to hickjack this issue for something differemt than the initial issue.
